### ISSUE TITLE
Chahub remove UNPUBLISHED competitions

### ISSUE THREAD

**Issue by ckcollab** [codalab/codalab-competitions#2266]
We recently encountered a problem where someone published a competition, that competition ended up on Chahub, only to be unpublished later. So, when a competition is unpublished we should let Chahub know.

We should add a `HideableMixin` something like this...

```python
# in Chahub
class HideableMixin(object):
    hidden = models.BooleanField(default=False)
    def save(self, *args, **kwargs):
        # if hidden somehow REMOVE from ElasticSearch and prevent it from re-adding?
        return super yadda yadda
```

Looking at the django elasticsearch plugin, we can maybe get similar behavior by overriding this to ignore `hidden`?

```python
class DocType(object):
    def get_queryset(self):
        """
        Return the queryset that should be indexed by this doc type.
        """
        return self._doc_type.model._default_manager.all()
```

**Comment by ckcollab** [comment]
I think it might be better to say `unpublished` instead of hidden?

**Comment by Tthomas63** [comment]
I think this issue should be resolved. Since we added the published field, anytime that changes on v1.5 we should get an update with that field changing on Chahub. The search shouldn't list it anymore.

**Comment by ckcollab** [comment]
Chahub determines whether or not to display unpublished competitions now, so we send it all over
